src/armature/GuidingVectors.ts

```diff
@@ -183,45 +191,147 @@ export class GuidingVectors implements CostFn {
 
             // Find the closest point on a guiding curve
             const closest = this.closest(parentPosition);
+            lastClosest = closest;
+
+            totalCost += this.computeCost(closest, vector, vec3From4(parentPosition));
+        });
+
+        let heuristicCost = 0;
+        if (lastClosest !== null) {
+            instance.getSpawnPoints().forEach((spawnPoint: SpawnPoint) => {
+                heuristicCost += this.computeCost(
+                    this.getOrCreateSpawnPointClosest(spawnPoint),
+                    this.getOrCreateSpawnPointVector(instance.generator, spawnPoint),
+                    <vec3>lastLocation
+                );
+            });
+        }
+
+        return { realCost: totalCost, heuristicCost };
+    }
+
+    /**
+     * Computes the cost for a given piece of added structure.
+     *
+     * @param {Closest} closest Information about the curve closest to the base of the parent.
+     * @param {vec4} added The direction vector for the added structure.
+     * @param {vec3} parentPosition The position of the parent that the added structure was
+     * attached to.
+     * @returns {number} The cost, given the above.
+     */
+    private computeCost(closest: Closest, added: vec4, parentPosition: vec3): number {
+        const distance = vec3.distance(Mapper.coordToVector(<coord>closest.point), parentPosition);
+
+        // Evaluate distance cost polynomial using Horner's method
+        let distanceCost = 0;
+        for (let power = 2; power >= 0; power -= 1) {
+            distanceCost = closest.curve.distanceMultiplier[power] + distanceCost * distance;
```

Comment:
It does get multiplied, but in the next iteration. (So the only one that doesn't get multiplied is the 0th entry, since it's the last one in the loop, which corresponds to x^0 anyway so it doesn't need to be multiplied.) Given that this is a small polynomial and we have more pressing concerns, maybe using Horner's method to evaluate the polynomial is overkill since it's harder to read, I guess.